jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/AsyncContentProducer.java

```diff
@@ -303,197 +257,71 @@ boolean isUnready()
-            {
-                if (chunk instanceof Content.Chunk.Error errorChunk)
-                {
-                    // Set the _error flag to mark the chunk as definitive, i.e.:
-                    // do not try to produce new raw chunk to get a fresher error
-                    // when the terminal chunk was generated by the interceptor.
-                    _error = true;
-                    if (_servletChannel.getResponse().isCommitted())
-                        _servletChannel.abort(errorChunk.getCause());
-                }
-
-                if (LOG.isDebugEnabled())
-                    LOG.debug("interceptor generated terminal chunk {}", this);
-            }
-            else if (chunk != _rawChunk && !_rawChunk.isTerminal() && _rawChunk.hasRemaining() && _rawChunk.remaining() == remainingBeforeInterception)
-            {
-                IOException failure = new IOException("Interceptor " + _interceptor + " did not consume any of the " + _rawChunk.remaining() + " remaining byte(s) of chunk");
-                if (chunk != null)
-                    chunk.release();
-                consumeCurrentChunk();
-                // Set the _error flag to mark the chunk as definitive, i.e.:
-                // do not try to produce new raw chunk to get a fresher error
-                // when the terminal chunk was caused by the interceptor not
-                // consuming the raw chunk.
-                _error = true;
-                Response response = _servletChannel.getResponse();
-                if (response.isCommitted())
-                    _servletChannel.abort(failure);
-                if (LOG.isDebugEnabled())
-                    LOG.debug("interceptor did not consume chunk {}", this);
-                chunk = _transformedChunk;
-            }
-
-            if (LOG.isDebugEnabled())
-                LOG.debug("intercepted raw chunk {}", this);
-            return chunk;
-        }
-        catch (Throwable x)
-        {
-            IOException failure = new IOException("bad chunk", x);
-            consumeCurrentChunk();
-            // Set the _error flag to mark the chunk as definitive, i.e.:
-            // do not try to produce new raw chunk to get a fresher error
-            // when the terminal chunk was caused by the interceptor throwing.
-            _error = true;
-            Response response = _servletChannel.getResponse();
-            if (response.isCommitted())
-                _servletChannel.abort(failure);
-            if (LOG.isDebugEnabled())
-                LOG.debug("interceptor threw exception {}", this, x);
-            return _transformedChunk;
         }
     }
 
-    private Content.Chunk produceRawChunk()
+    private Content.Chunk readChunk()
     {
         Content.Chunk chunk = _servletChannel.getServletContextRequest().read();
         if (chunk != null)
         {
-            _rawBytesArrived += chunk.remaining();
+            _bytesArrived += chunk.remaining();
```

Comment:
It might be a tiny bit more efficient to use _bytesArrived > 0 in the test below for setting the first byte time.

Reply:
@gregw I think there are cases where the chunk may be empty, for example `EOF` or `Trailers`, so I think it's more correct to compare to `MIN_VALUE`.